Azure Monitor: Basic Logs Backend

jcolladokuri opened the pull request:
<!--

Thank you for sending a pull request! Here are some tips:

1. If this is your first time, please read our contribution guide at https://github.com/grafana/grafana/blob/main/CONTRIBUTING.md

2. Ensure you include and run the appropriate tests as part of your Pull Request.

3. In a new feature or configuration option, an update to the documentation is necessary. Everything related to the documentation is under the docs folder in the root of the repository.

4. If the Pull Request is a work in progress, make use of GitHub's "Draft PR" feature and mark it as such.

5. If you can not merge your Pull Request due to a merge conflict, Rebase it. This gets it in sync with the main branch.

6. Name your PR as "<FeatureArea>: Describe your change", e.g. Alerting: Prevent race condition. If it's a fix or feature relevant for the changelog describe the user impact in the title. The PR title is used to auto-generate the changelog for issues marked with the "add to changelog" label.

7. If your PR content should be added to the What's New document for the next major or minor release, add the **add to what's new** label to your PR. Note that you should add this label to the main PR that introduces the feature; do not add this label to smaller PRs for the feature.

-->

**What is this feature?**

Adds backend support in Azure Monitor Logs service for the `search` API so that Basic Logs queries can be run from the front end

This PR also fixes a minor bug in the logic for showing the basic logs toggle that was causing the toggle to always reset to Analytics if the visualization was changed

**Why do we need this feature?**

For frontend Basic Logs implementation to work



Fixes #84950 #84948 #84951

aangelisc commented on pkg/tsdb/azuremonitor/azuremonitor.go:
```diff
@@ -19,6 +19,7 @@ import (
 	"github.com/grafana/grafana-plugin-sdk-go/backend/log"
 	"github.com/grafana/grafana-plugin-sdk-go/backend/resource/httpadapter"
 
+	ngalertmodels "github.com/grafana/grafana/pkg/services/ngalert/models"
```
We can't import this as it's core Grafana code, we'll have to hardcode the header

jcolladokuri replied:
changed to a variable within the function :)

aangelisc commented on pkg/tsdb/azuremonitor/loganalytics/azure-log-analytics-datasource_test.go:
```diff
@@ -91,12 +91,14 @@ func TestBuildingAzureLogAnalyticsQueries(t *testing.T) {
 		{
-			name: "Query with macros should be interpolated",
+			name:      "Query with macros should be interpolated",
+			fromAlert: true,
```
Why is `fromAlert` true for this test?

aangelisc commented on pkg/tsdb/azuremonitor/loganalytics/azure-log-analytics-datasource_test.go:
```diff
@@ -214,7 +218,8 @@ func TestBuildingAzureLogAnalyticsQueries(t *testing.T) {
 		{
-			name: "Queries with multiple resources",
+			name:      "Queries with multiple resources",
+			fromAlert: true,
```
Same question here 😊

aangelisc commented on pkg/tsdb/azuremonitor/loganalytics/azure-log-analytics-datasource_test.go:
```diff
@@ -254,7 +259,8 @@ func TestBuildingAzureLogAnalyticsQueries(t *testing.T) {
 		{
-			name: "Query with multiple resources",
+			name:      "Query with multiple resources",
+			fromAlert: true,
```
And here

aangelisc commented on pkg/tsdb/azuremonitor/loganalytics/azure-log-analytics-datasource_test.go:
```diff
@@ -1450,8 +1591,11 @@ func TestBuildingAzureLogAnalyticsQueries(t *testing.T) {
-			queries, err := datasource.buildQueries(ctx, tt.queryModel, dsInfo)
+			queries, err := datasource.buildQueries(ctx, tt.queryModel, dsInfo, tt.fromAlert)
 			tt.Err(t, err)
+			if err != nil {
```
I'd rather not return so if any other tests error we see all the errors at once, wdyt?

jcolladokuri replied:
Yes! Fair point, I have removed the logic to return from the function if there are any errors.

aangelisc commented on pkg/tsdb/azuremonitor/loganalytics/utils.go:
```diff
@@ -31,3 +34,28 @@ func AddConfigLinks(frame data.Frame, dl string, title *string) data.Frame {
+// Check whether a query should be handled as basic logs query
+// 1. basic logs flag is true
+// 2. resource selected is a workspace
+// 3.  query is not an alerts query
```
```suggestion
// 3. query is not an alerts query
```

aangelisc commented on pkg/tsdb/azuremonitor/loganalytics/utils.go:
```diff
@@ -31,3 +34,28 @@ func AddConfigLinks(frame data.Frame, dl string, title *string) data.Frame {
+	}
+
+	if fromAlert {
+		return false, fmt.Errorf("basic Logs queries cannot be used for alerting")
```
```suggestion
		return false, fmt.Errorf("basic Logs queries cannot be used for alerts")
```

aangelisc commented on pkg/tsdb/azuremonitor/loganalytics/azure-log-analytics-datasource.go:
```diff
@@ -144,6 +155,12 @@ func (e *AzureLogAnalyticsDatasource) buildQueries(ctx context.Context, queries
 				resourceOrWorkspace = *azureLogAnalyticsTarget.Workspace
 			}
 
+			meetsBasicLogsCriteria, meetsBasicLogsCriteriaErr := MeetsBasicLogsCriteria(basicLogsQueryflag, resources, fromAlert)
```
I suggest restructuring this so it only runs if `basicLogsQueryFlag` is `true` to avoid running this for every query

jcolladokuri replied:
I changed this! Let me know what you think

aangelisc commented on pkg/tsdb/azuremonitor/loganalytics/azure-log-analytics-datasource.go:
```diff
@@ -129,10 +134,16 @@ func (e *AzureLogAnalyticsDatasource) buildQueries(ctx context.Context, queries
 				resultFormat = types.TimeSeries
 			}
 
+			basicLogsQueryflag := false
```
```suggestion
			basicLogsQueryFlag := false
```

aangelisc reviewed (commented):
Excellent work @jcolladokuri! Just some comments to be addressed 😊

owensmallwood committed:
Unified Storage: Adds health check service for entity server (#86471)

owensmallwood committed:
Unified Storage: Redacts logged env var values for entity api db password (#87191)

redacts logged env var values for entity api db password

adela-almasan committed:
XYChart: Fix default point size (#87192)

imatwawana committed:
Docs: Add data links shared content (#86893)

* Added data link options shared file and added file to relevant visualizations

* Updated some text and links in canvas

* Added shared file to xy chart

* Fixed list of visualizations using shared file

* Removed unneeded level offset

* Removed future tense

* Update data links description

* Updated wording

* Removed redundant sentence

imatwawana committed:
Docs: add field overrides shared content (#87197)

* Added field overrides shared file

* Added link to overrides file in visualizations pages

* ihm/010524-field-override-shared-content/ run lint

---------

Co-authored-by: jev forsberg <jev.forsberg@grafana.com>

imatwawana committed:
Docs: add standard options shared content (#87199)

* Added standard options shared file

* Added missing option to shared file and added shared file link to visualizations pages

* Fixed links

jackw committed:
Chore: Add back node-forge npm package (#87136)

zserge committed:
Chore: Moving to slog (folders) (#87000)

* moving folders to slog

* trying to fix the tests

* fix per-logger filters in slog

* allow using slog.Default()

* bring cfg back to keep the pr small

* fix tests

* back to the roots

tskarhed committed:
AnnotationList: Fix link for annotation with no panel or dashboard (#87048)

* Handle case for no associated panel or dashboard

* Convert to CSS object

* Add tests and extend type to cover null use case

academo committed:
FE Sandbox: Fix innerHTML distortion causing issues with DOMPurify inner works (#87117)

* FE Sandbox: Fix innerHTML distortion causing issues with DOMPurify inner works

* Add a comment explaining the fix

jackw committed:
Fix: Always fetch plugin css with Systemjs (#87211)

* fix(plugins): fetch strategy should always be used for css/json/wasm/html

* refactor(plugins): prefer checking for js file type rather than list of other file types

IevaVasiljeva committed:
RBAC: Do not set permissions on data sources with wildcard UID in OSS (#87220)

do not set permissions on DS with wildcard UID

gabor committed:
postgres: decouple the frontend-part (#87131)

github-actions[bot] committed:
I18n: Download translations from Crowdin (#87170)

New Crowdin translations by GitHub Action

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

joshhunt committed:
E2C: Restore Disconnect modal (#87223)

zserge committed:
Chore: Remove cfg from folder service (#87212)

remove cfg from folder service

gillesdemey committed:
Alerting: Prevent search from locking the browser (#87132)

Sergej-Vlasov committed:
DashboardScene: Remove min height on viz picker scroll container (#87172)

removed min height prop

Sergej-Vlasov committed:
DashboardQueryEditor: Adjust options filtering in scenes (#87174)

adjust options filtering for dashboard datasource in scenes

mildwonkey committed:
Chore: update go.mod (#87194)

* Chore: update go.mod

scottlepp committed:
fix: sql expressions - sql parser table name case (#87196)

* fix: sql parser table name case

kalleep committed:
Identity: Use typed namespace id (#87121)

* Use typed namespace id

dmihai committed:
SSO: Update `ssoSettingsApi` flag to general availability (#87216)

* update ssoSettingsApi flag to general availability

* generate files

kevinwcyu committed:
CloudWatch: Fix apostrophes in dimension values not being escaped (#87182)

gamab committed:
RBAC: fix panic role not found permission sync (#87217)

Sergej-Vlasov committed:
DashboardScene: adjust lib panel as a dashboard datasource refresh logic (#87120)

* activate library viz panel and adjust id extractor

* revert parts that are fixed in #87125

* clean up after activation handler

dmihai committed:
SSO: Collect user save attempts from OAuth provider forms (#87233)

collect user save attempts from OAuth provider forms

axelavargas committed:
Dashboard Scene: Chore - Add missing unit test to PanelDataQueryTab (#87139)

* Create basic structure PanelDataQueriesTab unit test

* Add more test cases to the PanelDataQueriesTab test

* Update public/app/features/dashboard-scene/panel-edit/VizPanelManager.tsx

darrenjaneczek committed:
datatrails: include step index and number of steps when reporting trail step navigation usage (#87068)

* fix: include step index and number of steps

suntala committed:
Storage: Add mode 2 dual writing improvements (#87204)

* Fix mode 2 List test
* Set origin timestamp during conversion to k8s resource
* Add instructions for updating a playlist
* Handle partial deletions of a collection in mode 2

fayzal-g committed:
Alerting: Get grafana-managed alert rule by UID (#86845)

* Add auth checks and test

* Check user is authorized to view rule and add tests

* Change naming

* Update Swagger params

* Update auth test and swagger gen

* Update swagger gen

* Change response to GettableExtendedRuleNode

* openapi3-gen

* Update tests with refactors models pkg

Sergej-Vlasov committed:
DashboardScenes: Pin go version to go.mod file in scenes e2e workflow (#87238)

* adjust workflow to use correct go version

* finalise wokflow adjustments

gillesdemey committed:
Scenes: Add endLineNumber to json diff (#87049)

suntala committed:
Storage: Enable mode 2 dual writing for playlists (#87245)

Enable mode 2 dual writing for playlists

torkelo committed:
Select: Fixes issue preserving search term (input) when selecting a value (#87013)

ryantxu committed:
FeatureToggles: Write enabled flags to the logs on startup (#87225)

kaydelaney committed:
Scenes: Adjust styles to allow shrinking of variable selectors (#87085)

ashharrison90 committed:
Navigation: show `img` as backup if `icon` doesn't exist (#87247)

show img as backup if icon doesn't exist

gtk-grafana committed:
Logs: Explore panel default visualization feature flag (#87189)

* add logsExploreTableDefaultVisualization feature flag

* use feature flag in calculating initial visualization type

* add feature flag to tracking

L-M-K-B committed:
Renovate: Add Slate and related packages to ignore list (#87250)

refactor: add Slate and related dependencies to ignore section

NicholasPCole committed:
Docs: Fix link to values.yaml file (#87014)

Fix link to values.yaml file

Linking to the "edit" URL fails if the reader doesn't have write access
to the repo: they'll be prompted to fork the repo or to sign in to
GitHub. Instead link to the "blob" URL which is accessible to all
readers.

ashharrison90 committed:
Chore: Add tracking for total number of scss files (#87259)

* add tracking for total number of scss files

* need this comma :o

renovate[bot] committed:
Update React Aria (#87253)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
Update babel monorepo to v7.24.5

eleijonmarck committed:
TeamLBAC: Add scenario 6 to the docs (#87215)

* add task 6

* Apply suggestions from code review

Co-authored-by: Ieva <ieva.vasiljeva@grafana.com>

---------

Co-authored-by: Ieva <ieva.vasiljeva@grafana.com>

mdcruz committed:
docs: update flame graph visualization (#86615)

* docs: update flame graph visualization

* fix: linting issues

* docs: add example data

* Apply suggestions from code review

Co-authored-by: Isabel Matwawana <76437239+imatwawana@users.noreply.github.com>

* Removed Wikipedia link

* Fixed punctuation

* Updated max image width

* Update docs/sources/panels-visualizations/visualizations/flame-graph/index.md

Co-authored-by: Isabel Matwawana <76437239+imatwawana@users.noreply.github.com>

* fix: linting issues

* docs: add links to pyroscope docs about flame graph and profile types

* Ran prettier

* docs: add flame graph video

* Updated wording

---------

Co-authored-by: Isabel Matwawana <76437239+imatwawana@users.noreply.github.com>
Co-authored-by: Isabel Matwawana <isabel.matwawana@grafana.com>

imatwawana committed:
Docs: add panel options shared content (#87258)

* Added panel options shared file

* Added panel options shared file to visualizations pages

* Fixed version interpolation syntax and added Cloud links

* Fixed heading level of title

drew08t committed:
Canvas: Fix datalinks using field variable (#87265)

scottlepp committed:
fix: sql expressions parser with join (#87263)

NWRichmond committed:
Prometheus: Ensure values in metric selector are visible (#87150)

* fix: ensure selectable values are visible

* refactor: remove unnecessary ternaries, strengthen types

* docs: fix typo

* style: comply with `betterer` guidelines

* refactor: add clarity

* fix: remove border for consistency's sake

tomglenn committed:
docs: Minor tweaks to the Grafana Fundamentals tutorial (#86777)

* Minor tweaks to the Grafana Fundamentals tutorial

* Replaced old note style with admonition shortcode

* trigger CI

* Fixed shortcode

Co-authored-by: Danielle Broadwater <danibroadwater@outlook.com>

---------

Co-authored-by: Isabel Matwawana <76437239+imatwawana@users.noreply.github.com>
Co-authored-by: Isabel Matwawana <isabel.matwawana@grafana.com>
Co-authored-by: Danielle Broadwater <danibroadwater@outlook.com>
Co-authored-by: Jack Baldry <jack.baldry@grafana.com>

renovate[bot] committed:
Update dependency @floating-ui/react to v0.26.13 (#87273)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
Update dependency @grafana/scenes to v4.13.2

renovate[bot] committed:
Update dependency @kusto/monaco-kusto to v10.0.20

kalleep committed:
IDToken: add namespace (#87242)

Clarity-89 committed:
Grafana/ui: Update Button and IconButton components to accept icon as component (#87243)

* Saga-icons: Forward SVG attributes

* Icon prop as component

* Add IconRenderer

* Cleanup

* Fix types

* Update size prop

* Revert storybook

* Update IconButton

bergquist committed:
Scopes: Adds kinds for browsing the scope node tree.  (#86975)

Signed-off-by: bergquist <carl.bergquist@gmail.com>

github-actions[bot] committed:
I18n: Download translations from Crowdin (#87240)

New Crowdin translations by GitHub Action

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

chaudyg committed:
k8s: remove dependency on app_mode development (#85474)

* k8s: remove dependency on app_mode development

* remove dev conditions on storage server

* remove logic to drop entity tables

renovate[bot] committed:
Update dependency @pmmmwh/react-refresh-webpack-plugin to v0.5.13 (#87283)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
Update dependency @react-types/button to v3.9.3

kaydelaney committed:
Dashboard: Fix issue where changes to panel in row weren't detected in panel edit (#87195)

Closes #87171

renovate[bot] committed:
Update dependency @react-types/menu to v3.9.8

eleijonmarck committed:
Auditlogging: add docs about dsquery (#87218)

* add docs for dsquery

* update the generic actions

renovate[bot] committed:
Update dependency @react-types/overlays to v3.8.6

itsmylife committed:
InfluxDB: Support cardinality queries with backend mode (#87264)

support cardinality queries

renovate[bot] committed:
Update dependency @swc/helpers to v0.5.11 (#87296)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
Update dependency @types/node to v20.12.8

domasx2 committed:
Tempo: Option to not show variable options in "search" editor (#87126)

* allow disabling variable options in Tempo "search" editor

* also for GroupByField

* set addVariablesToOptions true by default

gillesdemey committed:
Alerting: Fix max alerts (#87271)

gillesdemey committed:
Alerting: Add OAuth2 to HTTP settings for vanilla Alertmanager / Mimir (#87272)

add OAuth2 to HTTP settings for vanilla Alertmanager / Mimir

renovate[bot] committed:
Update dependency dompurify to v3.1.2

andresmgot committed:
Migrate wrong datasource UIDs (#86598)

santihernandezc committed:
Alerting: Implement GetStatus in the remote Alertmanager struct (#84887)

* Alerting: Implement GetStatus in the remote Alertmanager struct

* update tests

* fix tests, extract AlertmanagerConfig from PostableConfig

* get the remote AM config instead of the Grafana one from the remote AM

* pass grafana AM config in test

* return error in GetStatus instead of logging it (internal AM)

renovate[bot] committed:
Update dependency i18next to v23.11.3

scottlepp committed:
sql expressions: improve parser (#87277)

sql expressions: improve parser

renovate[bot] committed:
Update dependency lru-cache to v10.2.2

renovate[bot] committed:
Update dependency re-resizable to v6.9.16

itsmylife committed:
Datatrails: Fix overflow of labels (#87301)

Fix overflow of labels

renovate[bot] committed:
Update dependency rudder-sdk-js to v2.48.7

csmarchbanks committed:
Prometheus: Add reqAction to Prometheus routes (#87208)

Add reqAction to Prometheus routes

This should allow provisioned service accounts that do not have a role to use the /proxy endpoints for Prometheus. This
is desired for some integrations that need to query for labels or other data not part of /ds/query.

renovate[bot] committed:
Update dependency yaml to v2.4.2

aangelisc committed:
MSSQL: Add `HostName` method (#87309)

* Add HostName method

* Add test

* Fix lint

onprem committed:
Storage Server: Avoid incorrect parsing of connection string (#85293)

Storage Server: avoid incorrect parsing of connection string

In storage server, a connection string is generated to connect to the
backing database. In case of Postgres, we were generating the string
like `"user=grafana pass= host=postgres.example port=4567"`.

This triggered an edge case in `pq` (the go postgres driver) to parse
`pass` to be equal to `host=postgres.example`, and host being reset to
default value.

Using single quotes in the connection string fixed this.

Signed-off-by: Prem Kumar <prem.saraswat@grafana.com>

juanicabanas committed:
ShareModal: Share link redesign under `newDashboardSharingComponent` FF (#87011)

renovate[bot] committed:
Update Yarn to v4.2.1 (#87325)

* Update Yarn to v4.2.1

* add new yarn version

---------

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>
Co-authored-by: Ashley Harrison <ashley.harrison@grafana.com>

Sergej-Vlasov committed:
DashboardScene: Restrict dashboard in read only state (#87316)



---------

Co-authored-by: Ivan Ortega Alba <ivanortegaalba@gmail.com>

darrenjaneczek committed:
datatrails: recently loaded trails modify parent variables when making new steps (#87284)

* fix: loaded trails modified parent var on new step

This ensures that recently loaded trails won't have variable changes
which create new steps modify the previous step.

gillesdemey committed:
Alerting: New settings page (#84501)

iwysiu committed:
Cloudwatch: Update grafana-aws-sdk to fix sts endpoints (#87326)

Cloudwatch: update grafana-aws-sdk to fix sts endpoints

darrenjaneczek committed:
scenes: update to v4.14.0: opt-in to `useQueriesAsFilterForOptions` (#87244)

* fix: opt-in to `useQueriesAsFilterForOptions`

With updating the scenes library, dashboards must now explicitly
opt in to `useQueriesAsFilterForOptions`.

L-M-K-B committed:
Dashboards: Add toggle for folder names in dashboard list panel (#87288)

* refactor: add switch

* refactor: add functionality

* refactor: fix test

renovate[bot] committed:
Update dependency @grafana/faro-core to v1.7.0 (#87329)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

ashharrison90 committed:
Chore: improve typings (#87306)

* some fixes

* some more fixes

* some canvas fixes

* use removeProperty instead

* few more

* fixy fix

* undo changes to ticks for now

darrenjaneczek committed:
datatrails: quick bugfix: handle undefined step (#87337)

fix: handle undefined step

darrenjaneczek committed:
datatrails: only fetch metrics metadata when using the language provider, avoid redundant fetch of metric names (#87067)

* fix: only fetch metrics metadata, not metric names

renovate[bot] committed:
Update dependency @grafana/faro-web-sdk to v1.7.0 (#87339)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

Clarity-89 committed:
PanelHeaderCorner: Remove font-awesome icons (#87303)

* PanelHeaderCorner: Remove font-awesome

* Fix mode styles

* Remove unused styles

* Cleanup

* Simplify styles

* Fixes

* Remove unused variable

darrenjaneczek committed:
datatrails: reloaded trail bugs related to no selected metric (#87344)

* fix: reloaded scene first step topScene

The top scene was not being initialized
when reloading trails, causing it to use
the top scene of whatever was selected
previously.

* fix: select new metric node not reloading right

On reloading trails, any metric node that
was 'green' with a undefined metric due to
clicking on the "select new metric" button,
was reloading with the metric of the step
directly before it (in index order).

JacobsonMT committed:
Alerting: Hook up GMA silence APIs to new authentication handler (#86625)

This PR connects the new RBAC authentication service to existing alertmanager API silence endpoints.

charandas committed:
ext_jwt: switch to new authlib (#87157)

imatwawana committed:
Docs: add data link cell type information (#87350)

Add data link cell type information

nmarrs committed:
Canvas: Fix phantom connections persisting after element deletion (#87285)

adela-almasan committed:
XyChart: Fix legend (#87155)

adela-almasan committed:
XY Chart: Prevent crash when deleting a series  (#87356)

nmarrs committed:
Canvas: Add support to rotate a group of elements (#87358)

nmarrs committed:
Canvas: Fix canvas style regression (#87363)

JonCole committed:
Azure: get custom cloud list from grafana-azure-sdk-go package (#86717)

* Get custom cloud list from grafana-azure-sdk-go for frontend settings

* Remove getAzureClouds() and call CustomClouds() directly

* remove unused type

ryantxu committed:
Prometheus: Avoid parsing unused legacy datasource property (#87224)

* query cleanup

* parse query

* more comments

mgyongyosi committed:
Auth: Add OrgRoleMapper service (#86973)

* Add OrgRoleMapper

* Address feedback, add more tests

* Prevent resetting assignments when global org mapping fails

* Provide a parsing function instead of recalculating the org_mapping on every mapping

* Introduce strict role parsing/mapping

* Introduce MappingConfiguration

* Handle other edge case

* Add tests

* lint

* Apply documentation update suggestions

* Apply GetDefaultOrgMapping suggestions from code review

* Apply suggestions for cleaning up unnecessary err in ParseOrgMappingSettings

* Apply suggestions from code review

* Address feedback suggestions

* Cleanup

* Reduce cognitive complexity of ParseOrgMappingSettings

---------

Co-authored-by: Gabriel MABILLE <gamab@users.noreply.github.com>

lwandz13 committed:
Docs:  added note to RBAC doc from support escalation (#85311)

* added admonition and API info

* Update docs/sources/administration/roles-and-permissions/access-control/_index.md

---------

Co-authored-by: Misi <mgyongyosi@users.noreply.github.com>
Co-authored-by: Jack Baldry <jack.baldry@grafana.com>

fabrizio-grafana committed:
Tempo: Fix sorting for nested tables (#87214)

eleijonmarck committed:
Actionsets: Fix downgrade of permission to add the actionset permission (#87328)

* check for missingActions empty and add actionset

* spelling

brendamuir committed:
Alerting vale fixes (#87380)

aocenas committed:
Flamegraph: Add diff mode color legend (#87319)

yuri-rs committed:
Loki: Use label/<name>/values API instead of series API for label values discovery (#83044)

* Loki: Add label values API selector to setting and use label_values API when selected

* remove CHANGELOG change

* add docs

* Support Loki 2.7+ only, so unconditionally use /labels API

* Correct doc for fetchLabels and fetchLabelValues functions

* Fixes after merge

* Do not encode query parameter twice

* return getLabelKeys in Completion Provider

* docs

* Add test for LabelParamEditor

* Update public/app/plugins/datasource/loki/LogContextProvider.test.ts

Co-authored-by: Ivana Huckova <30407135+ivanahuckova@users.noreply.github.com>

* toHaveBeenCalledWith

---------

Co-authored-by: Matias Chomicki <matyax@gmail.com>
Co-authored-by: Ivana Huckova <30407135+ivanahuckova@users.noreply.github.com>

gelicia committed:
Explore: Only let nodeGraph frames process through nodeGraph visualization (#87198)

* Only let nodeGraph frames process through nodeGraph

* add metadata to test frames to be processed

* Revert "add metadata to test frames to be processed"

This reverts commit fc366a0ab4928751b2382f7a09d41ead34bc9c89.

* Revert "Only let nodeGraph frames process through nodeGraph"

This reverts commit b8be50e2196410fc8155aed7630bfddd1408b62c.

* Only send node graph frames to node graph logic

* remove unnecessary logic

gillesdemey committed:
Alerting: Allow editing of provisioned mutable Alertmanagers (#87385)

github-actions[bot] committed:
I18n: Download translations from Crowdin (#87336)

New Crowdin translations by GitHub Action

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

ppcano committed:
Alerting docs: update Comparison between alert rule types (#87252)

ppcano committed:
Alerting docs: Update and restructure `Introduction/Alert rule evaluation`  (#87331)

* Migrate and update `Alert rule evaluation`

* Minor change

* Update internal links to new URLs

NWRichmond committed:
Prometheus: Place custom inputs first when using regex filter values in the query builder (#87360)

* fix: place custom inputs first when using regex filter values

* fix: remove confusing messaging

yincongcyincong committed:
fix ipv6 startup fail #81870 (#87108)

* fix ipv6 startup fail #81870

* ipv6 startup fail

* ipv6 startup fail

DanCech committed:
Chore: Fix error handling in postDashboard, remove UserDisplayDTO, fix live redis client initialization (#87206)

* clean up error handling in postDashboard and remove UserDisplayDTO

* replace GetUserUID with GetUID and GetNamespacedUID, enforce namespace constant type

* lint fix

* lint fix

* more lint fixes

Commit:
Revert #86466 (#87405)

Revert "Apply plugin route ReqAction to ds_proxy authorization (#86466)"

This reverts commit 53f94ac50dde7bc6c25f6a8254e85a2e8b1ae138.

leonorfmartins committed:
Choose mode. Turn DualWriter into a real interface. Better Logging (#87291)

* Choose mode. Add log field on top level dualwriter

* Add logs

* Turn DualWriter into a full interface. Fix tests

* Lint

* Use struct for dualWriter interface

* Use struct

* Default should be legacyStore for all entities

* Fix test. Get rid of extra concrete type

* Remove comment

* Add comment

* Temp set dualwriter mode 2 for playlists while configs are not in place

* Add modes type + add comment on what each mode does

* Don't require watcher interface for now

* Use storage implementation on mode 2

* Update pkg/apiserver/rest/dualwriter_mode2.go

Co-authored-by: Todd Treece <360020+toddtreece@users.noreply.github.com>

* Pass log values to the context

* test

* Update pkg/apiserver/rest/dualwriter_mode3.go

Co-authored-by: Dan Cech <dcech@grafana.com>

---------

Co-authored-by: Todd Treece <360020+toddtreece@users.noreply.github.com>
Co-authored-by: Dan Cech <dcech@grafana.com>

gabor committed:
mysql: fix dispose-method not called in external-mode (#87330)

tonypowa committed:
alerting: config notif page improvement (#87140)

* alerting: config notif page improvement

* Update docs/sources/alerting/configure-notifications/template-notifications/_index.md

Co-authored-by: brendamuir <100768211+brendamuir@users.noreply.github.com>

* Update _index.md

---------

Co-authored-by: brendamuir <100768211+brendamuir@users.noreply.github.com>

renovate[bot] committed:
Update dependency @grafana/faro-web-sdk to v1.7.2 (#87341)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
Update dependency @types/diff to v5.2.1

ashharrison90 committed:
Chore: replace calls to `defaultProps` (#87310)

* replace calls to defaultProps

* make linksGetter optional

kaydelaney committed:
Chore: Rewrite grafana-prometheus css using object styles (#86835)

kaydelaney committed:
Chore: Fix lint error in grafana-prometheus (#87421)

kaydelaney committed:
Chore: Rewrite alerting css using object styles (#87114)

Co-authored-by: Gilles De Mey <gilles.de.mey@gmail.com>

kalleep committed:
IDToken: Set typ header (#87430)

TokenSigning: Set typ header

oscarkilhed committed:
Dashboard scenes: Fix view panel breadcrum when going to edit from view single panel (#87411)

Fix view panel breadcrum when going to edit from view panel

tomratcliffe committed:
Chore: Allow MenuItems to have `role=menuitem` when URL is set (#86886)

Allow passing a role into a MenuItem component, and fallback to default behaviour if not passed

ivanahuckova committed:
Logs panel: Fix spacing when log details disabled (#87431)

* Logs panel: Fix spacing when log details disabled

* Fix error spacing

leonorfmartins committed:
Force interface implementation also on legacy storage (#87414)

* Force interface implementation also on legacy storage

* Add DeleteCollection to folders and dashboards

* Fix integration tests

* Fix tests

alexanderzobnin committed:
Access control: Cache basic roles and teams permissions (#87043)

* RBAC: Cache basic roles permissions

* Cache teams permissions

* Set cache TTL to 1 minute

* Add OSS implementation

* Fetch basic role permissions correctly

* fix conflict_user_command

* Fix teams permissions query

* Add traces for GetUserPermissions

* Fix folders tests

* Fix colflict user command

* Update store mock

* Fix linter error

* Reuse GetUserPermissions for fetching basic roles

* tests for GetTeamsPermissions

* pre-allocate slice capacity

* Fix linter

IevaVasiljeva committed:
Docs: Link to generic oauth docs from team sync page (#87381)

link to generic oauth docs from team sync page

eleijonmarck committed:
Actionsets: fix remove printf (#87439)

fix: remove printf

sympatheticmoose committed:
Plugins: Catalog to show all plugins by default (#87168)

* update plugin install filter to show all by default

* make pretty

* remove aria-label on lock icon

* wip: attempt to fix overflow

* switch to stack, add aria label back for testing

* make pretty

* switch to TextLink

* remove learn more link from catalog

* add hover text

* docs: update plugin install to reflect state filter change

* Revert "docs: update plugin install to reflect state filter change"

This reverts commit 16222aee0bd6960f768e810ef2c8686140849ef2.

ashharrison90 committed:
SCSS: remove some low hanging dead fruit (#87260)

* remove some low hanging fruit

* restore json_explorer

kalleep committed:
IDToken: Reuse claims from authlib (#87437)

* bump authlib version

* Reuse claims from authlib

wasim-nihal committed:
Auth: Added support to filter for parent teams in GitHub connector's team membership filter (#86754)

* added changes for parent teams in team_ids. see https://github.com/grafana/grafana/issues/85916

Signed-off-by: Syed Nihal <syed.nihal@nokia.com>

* added unit test

Signed-off-by: Syed Nihal <syed.nihal@nokia.com>

* addressed review comments to consider case where parent object can be null

Signed-off-by: Syed Nihal <syed.nihal@nokia.com>

* addressed review comment

Signed-off-by: Syed Nihal <syed.nihal@nokia.com>

---------

Signed-off-by: Syed Nihal <syed.nihal@nokia.com>

alexweav committed:
Alerting: Do not retry rule evaluations with "input data must be a wide series but got type long" style errors (#87343)

add typed error for series must be wide, do not retry

charandas committed:
ext_jwt: streamline expected aud in access tokens and id tokens (#87401)

DanCech committed:
Provisioning: Look up provisioned folders by UID when possible (#87465)

look up provisioned folders by uid when possible

stephaniehingtgen committed:
Chore: Upgrade go to 1.22.3 (#87463)

charandas committed:
K8s: upgrade authlib with correct IDTokenClaims (#87461)

torkelo committed:
DashboardDataSource: Fixes issue where sometimes untransformed data could be returned  (#87433)

* DashboardDataSource: always return a result key

* Update

* Update

torkelo committed:
DashboardScene: Fixes editing transformations after toggling table view (#87397)

renovate[bot] committed:
Update dependency @grafana/scenes to v4.17.1 (#87490)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

tonypowa committed:
alerting/webhook doc update (#87492)

* alerting/webhook doc update

* text formath

* fixed admonition

* typo

* headings

* formatted headings and prettified it all

* one more heading

kalleep committed:
Identity: Use typed version of namespace id  (#87257)

* Remove different constructors and only use NewNamespaceID

* AdminUser: check typed namespace id

* Identity: Add convinient function to parse valid user id when type is either user or service account

* Annotations: Use typed namespace id instead

IevaVasiljeva committed:
RBAC: Don't set empty action sets (#87398)

* don't set empty action sets

* extend comment

renovate[bot] committed:
Update dependency rc-cascader to v3.25.0 (#86045)

* Update dependency rc-cascader to v3.25.0

* fix type errors

---------

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>
Co-authored-by: Ashley Harrison <ashley.harrison@grafana.com>

renovate[bot] committed:
Update dependency @types/jquery to v3.5.30

tskarhed committed:
Revert "Panel: Keyboard focus shortcuts prioritization" (#87213)

Revert "Panel: Keyboard focus shortcuts prioritization (#86772)"

This reverts commit 3e450ec4bff3f971f6b885b6282bce8055a02a1b.

harisrozajac committed:
Explore: Logs in content outline (#86431)

* wip filtering

* Remove corresponding children from parentlessItem ref on register

* wip filtering

* Remove corresponding children from parentlessItem ref on register

* Highlighting filters

* OMG

* Handle updating highlight field inside context

* Toggling legend from the content outline

* Support multiple selects for log filters in content outline

* Remove commented out code

* Fix an issue with loki datasource; use same datastructure for logsVolume in Logs and LogVolume panel

* Fix registering logic

* cleanup

* Don't register outline items when having multiple log volume levels

* Remove checking mergeSingleChild logic because in context because ContentOutline does it

* Fix logic so it works for elasticsearch

* Better logic

* Add log color and sort

* Show items at full opacity

* Remove commented sort

* Do not highlight if all are selected, and highlight = 100% opacity, active = 30%

* dont auto collapse after filter unselect

* Remove console logs

* No partial opacity, add height for consistent display

---------

Co-authored-by: Piotr Jamroz <pm.jamroz@gmail.com>
Co-authored-by: Kristina Durivage <kristina.durivage@grafana.com>

joshhunt committed:
I18n: Fix and reenable verify-i18n CI check (#87496)

* I18n: Change verify-i18n script to use node_deb image with make in it

* remove swagger from i18n make

* make verify-i18n fail ci

* fix incorrect and unextracted nav translations

* fix more copy paste

JoaoSilvaGrafana committed:
GrafanaUI: Add `tabular` prop to Text component for tabular numbers (#87440)

github-actions[bot] committed:
I18n: Download translations from Crowdin (#87501)

New Crowdin translations by GitHub Action

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

renovate[bot] committed:
Update dependency @types/node to v20.12.11

axelavargas committed:
Chore: Update deprecated layout components HorizontalGroup and VerticalGroup in dashboard scene code (#87447)

* Chore: update deprecated layout components in scene code

* Add missing css for missing selectionOptionsForm

marefr committed:
Chore: Migrate to use buf for protobuf generation (#87407)

* Chore: Migrate to use buf for protobuf generation

* fix codeowners

* fixes after review

* docs/tooling for installing buf

* add back codeowners

* Update upgrading-dependencies.md

Co-authored-by: Giuseppe Guerra <giuseppe.guerra@grafana.com>

* sdk v0.229.0

---------

Co-authored-by: Giuseppe Guerra <giuseppe.guerra@grafana.com>

alexanderzobnin committed:
Access control: Fix searching permissions from cache (#87489)

* Fix searching permissions from cache

* Write permissions to cache

renovate[bot] committed:
Update dependency @types/prismjs to v1.26.4

renovate[bot] committed:
Update dependency @types/webpack-env to v1.18.5

toddtreece committed:
K8s: Add path rewriter filter (#87456)

renovate[bot] committed:
Update dependency rc-tree to v5.8.7

toddtreece committed:
K8s: Prevent the use of arbitrary namespaces (#83636)

renovate[bot] committed:
Update dependency react-hook-form to v7.51.4

renovate[bot] committed:
Update dependency semver to v7.6.1

toddtreece committed:
Scopes: Add a /find query endpoint (#87457)


---------

Co-authored-by: Ryan McKinley <ryantxu@gmail.com>

PoorlyDefinedBehaviour committed:
Ephemeral instances: update go version (#87526)

toddtreece committed:
GRPC Server: Add README (#87442)

adela-almasan committed:
Timeseries: Fix incorrect threshold style (#87521)

bossinc committed:
Azure Monitor: Update grafana-azure-sdk-go to v2.0.3 (#87480)

* update

* update

scottlepp committed:
Sql Expressions: use display name for sql tables (#87530)

diegommm committed:
Unified Storage: Add SQL template package (#87524)

* added sqltemplate package

* addded example

* fix linting issues

* improve code readability

* fix documentation

renovate[bot] committed:
Update Yarn to v4.2.2

ashharrison90 committed:
new yarn version

jackw committed:
Frontend: Improve barrel file detection in codemod (#87389)

IevaVasiljeva committed:
RBAC: Adding action set resolver for RBAC evaluation (#86801)

* add action set resolver

* rename variables

* some fixes and some tests

* more tests

* more tests, and put action set storing behind a feature toggle

* undo change from cfg to feature mgmt - will cover it in a separate PR due to the amount of test changes

* fix dependency cycle, update some tests

* add one more test

* fix for feature toggle check not being set on test configs

* linting fixes

* check that action set name can be split nicely

* clean up tests by turning GetActionSetNames into a function

* undo accidental change

* test fix

* more test fixes

tonypowa committed:
alerting: integrations > small update (#87538)

* added links to list

* changed heading

renovate[bot] committed:
Update dependency @grafana/scenes to v4.17.2

renovate[bot] committed:
Update dependency @opentelemetry/semantic-conventions to v1.24.1 (#87544)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

torkelo committed:
MixedDataSource: Support multi value data source variable that issues a query to each data source (#83356)

* MixedDataSource: Support multi value data source variable

* go work sync

* fix typo

* if it is not interpolated use the actual uid

* Update tests

* prettier

* go work sync

* add more unit test case

* Update tests to use scene variables

* merge main

* Update

---------

Co-authored-by: ismail simsek <ismailsimsek09@gmail.com>
Co-authored-by: Ryan McKinley <ryantxu@gmail.com>

renovate[bot] committed:
Update dependency @grafana/scenes to v4.17.3

stephaniehingtgen committed:
Chore: Upgrade go from 1.21.0 to 1.21.10 (#87479)

renovate[bot] committed:
Update dependency @playwright/test to v1.44.0 (#87550)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

kaydelaney committed:
Templating: Fix issue where updated custom var options weren't persisted (#86804)

* Templating: Fix issue where updated custom var options weren't persisted

toddtreece committed:
Chore: Add info about K8s folder structure to README (#87508)


---------

Co-authored-by: Ryan McKinley <ryantxu@gmail.com>

Sergej-Vlasov committed:
DashboardScene: Adjust failing scenes e2e (#87553)

* adjust failing scenes e2e

* remove .only from test

* restore e2e workflow

oscarkilhed committed:
DashboardScene: Discard panel changes disabled/enabled depending of changes (#87137)

---------

Co-authored-by: Ivan Ortega Alba <ivanortegaalba@gmail.com>
Co-authored-by: Dominik Prokop <dominik.prokop@grafana.com>

kylebrandt committed:
Scopes: Fix logic in find call (#87565)

With Find, if the parent query parameter is empty or not provided, we want to list the top level of the forest.

drew08t committed:
Canvas: Fix bug for connection first vertex (#87534)

baldm0mma committed:
Transformations: Fix true inner join in `joinByField` transformation (#87409)

* baldm0mma/inner_join_fix/ init commit

* baldm0mma/inner_join_fix/ first attempt at an tabular inner join

* baldm0mma/inner_join_fix/ add innerTabular

* baldm0mma/inner_join_fix/ add innerJoin option to editor

* baldm0mma/inner_join_fix/ add editor option

* baldm0mma/inner_join_fix/ update joinInnerTabular function behavior

* baldm0mma/inner_join_fix/ add js docs

* baldm0mma/inner_join_fix/ update jsdocs

* baldm0mma/inner_join_fix/ update docs

* baldm0mma/inner_join_fix/ remove unused console.logs

* baldm0mma/inner_join_fix/ update tests

* baldm0mma/inner_join_fix/ simplify getValue

* baldm0mma/inner_join_fix/ update tests

* baldm0mma/inner_join_fix/ update docs builder

* baldm0mma/inner_join_fix/ add tables to Outer join (for tabular data)

* baldm0mma/inner_join_fix/ update docs

* baldm0mma/inner_join_fix/ build docs

* baldm0mma/inner_join_fix/ remove innertab for inner

* baldm0mma/inner_join_fix/ rename innertab

* baldm0mma/inner_join_fix/ update tests

* baldm0mma/inner_join_fix/ rem con logs

* baldm0mma/inner_join_fix/ rem comment

* baldm0mma/inner_join_fix/ rem sample data

* baldm0mma/inner_join_fix/ remove irrelevant test

* baldm0mma/inner_join_fix/ update docs

* Update packages/grafana-data/src/transformations/transformers/joinDataFrames.test.ts

Co-authored-by: Nathan Marrs  <nathanielmarrs@gmail.com>

* baldm0mma/inner_join_fix/ update test description

---------

Co-authored-by: nmarrs <nathanielmarrs@gmail.com>

alexweav committed:
Alerting: Add database migration for recording rule fields (#87012)

* Create recording rule fields in model

* Add migration

* Write to database, support in version table

* extend fingerprint

* Force fields to be empty on validate

* Another storage spot, tests for fingerprint

* Explicitly set defaults in provisioning API

* Tests for main API validation

* Add diff tests even though fields are unpopulated for now

* Use struct tag approach instead of FromDB/ToDB hooks as it better handles nulls when deserializing

* test for deser

* Backout RecordTo for now since it's not decided in the doc

* back out of migration too

* Drop datasourceref for now

* address linter complaints

* Try a single outer struct with all fields embedded

yuri-tceretian committed:
Alerting: Add two sets of provisioning actions for rules and notifications  (#87149)

nmarrs committed:
Canvas: Chore clean up betterer (#87477)

timo committed:
SQL Query Editor: Fix label-for IDs, associate "Table" label (#86944)

* SQL Query Editor: Fix label-for IDs, associate "Table" label

* Fix mysql e2e test

---------

Co-authored-by: Zoltán Bedi <zoltan.bedi@gmail.com>

toddtreece committed:
Chore: Update go workspace check error message (#87576)

leandro-deveikis committed:
Caching: Implement mtls-enabled memcached integration (#84171)

* fix merge

* fix conflicts

* updating generated feature toggle code

* Fix go mod owner

* fix merge

* add dependency

* fix dependencies

* workspace sync

* go get

* work sync

* trying something

* fix

* add docs for tls config

* work sync

* adjust go mod

* try something

* move import to the correct file

---------

Co-authored-by: Michael Mandrus <michael.mandrus@grafana.com>

csmarchbanks committed:
Prometheus: Use correct datasources:write permission (#87454)

The datasources:edit permission does not exist and should have been write to begin with.

hairyhenderson committed:
ci: use go.mod as source of truth for actions/setup-go action (#87574)

* ci: use go.mod as source of truth for actions/setup-go action

Signed-off-by: Dave Henderson <dave.henderson@grafana.com>

* updating upgrading-go-version.md doc

Signed-off-by: Dave Henderson <dave.henderson@grafana.com>

---------

Signed-off-by: Dave Henderson <dave.henderson@grafana.com>

alexweav committed:
Alerting: Do not store series values from past evaluations in state manager for no reason (#87525)

Do not store previous execution results on states

gelicia committed:
Revert "MixedDataSource: Support multi value data source variable that issues a query to each data source" (#87579)

Revert "MixedDataSource: Support multi value data source variable that issues…"

This reverts commit 700d378c985070d70147185f15d7061e59f3e9d6.

drew08t committed:
Canvas: Improve connection vertex logic (#87580)

Canvas: Prevent machine error causing logic swap

nmarrs committed:
Canvas: Fix infinite panning editor description (#87582)

bossinc committed:
Azure data sources: Set selected config type before save (#87481)

* set selected config type before save

* use effect once and move in mssql

* add to prom

* Update ConfigurationEditor.tsx

* Update ConfigurationEditor.tsx

* Delete public/app/plugins/datasource/mssql/configuration/ConfigurationEditor.tsx

* omadsfdsa

* remove unneeded assign

mgyongyosi committed:
Chore: Implement GetOAuthProviders in FakeSocialService (#87590)

Implement GetOAuthProviders in FakeSocialService

alexanderzobnin committed:
Access control: Use search options for computing permissions cache key (#87589)

aocenas committed:
Flamegraph: Add collapse and expand group buttons to toolbar (#87395)

svennergr committed:
Loki: Fix log context when no label types are present (#87587)

ashharrison90 committed:
Chore: Remove some more old SCSS (#87506)

* remove some more old styles and consolidate any angular ones into their own file

* few more

soniaAguilarPeiron committed:
Alerting: Reduce number of request fetching rules in the dashboard view using rtkq (#86991)

* Reduce number of request fetching rules in the dashboard view using rtkq

* Fix UnifiedAlertStatesWorker work

* refactor ungroupRulesByFileName

* Address review comments and fix test

* fix DashboardQueryRunner test

* Fix tests

* Update AlertStatesDataLayer to use RTKQ

* Fix PanelAlertTabContent test

* Fix PanelAlertTabContent test after adding RTKQ

* fix test and address PR review comments

* Update useCombinedRuleNamespaces to have both dashboardUID and panelId as optional params and rename the hook

* Address review pr comment

* remove test about template variables

* Use poll interval in useCombinedRules

kalleep committed:
Chore:: Update authlib version (#87603)

update authlib version

IevaVasiljeva committed:
Chore: Remove use of deprecated method in AC code (#87541)

* switch from using cfg to using featuremgmt for checking a feature toggle in AC code

* merge test fixes

kaydelaney committed:
Chore: Use getNextRefId instead of deprecated getNextRefIdChar (#87460)

kaydelaney committed:
Chore: Remove deprecated re-exported template variable types (#87459)

Chore: Remove deprecated re-exported template varible types

renovate[bot] committed:
Update dependency systemjs to v6.15.1 (#87594)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
Update dependency @prometheus-io/lezer-promql to v0.52.0 (#87554)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
Update dependency eslint-plugin-jsdoc to v48.2.4

cwbriones committed:
Explore: lookup datasource by name when present in legacy URLs (#85222)

* [explore] lookup datasource by name when present in legacy compact URLs

* update unit test

* prettier fix

---------

Co-authored-by: Kristina Durivage <kristina.durivage@grafana.com>

linoman committed:
Chore: Add login protection when user is trying different uppercase letters (#87588)

* add login protection when the user is trying different uppercase letters

ivanahuckova committed:
Docs: Add Configuring Elasticsearch documentation for required privileges (#87593)

* Docs: Add Configuring Elasticsearch documentation for required privileges

* Update docs/sources/datasources/elasticsearch/configure-elasticsearch-data-source.md

* Update docs/sources/datasources/elasticsearch/configure-elasticsearch-data-source.md

Co-authored-by: Sven Grossmann <sven.grossmann@grafana.com>

---------

Co-authored-by: Sven Grossmann <sven.grossmann@grafana.com>

renovate[bot] committed:
Update dependency glob to v10.3.14

simonswine committed:
[DOC] Adds missing link to Pyroscope core data source (#85493)

[DOC] Add missing link to Pyroscope

Grafana Pyroscope is a core shipped data source.

torkelo committed:
Dropdown: Fixes double call to onVisibilityChange (#87607)

* Dropdown: Fixes double call to onVisibilityChange

* Update

* update

tskarhed committed:
TimeSeries: Improve keyboard focus and fix spacebar override (#86848)

* Make GrpahNG focusable

* Move to VizLayout instead

* Update uPlot plugins for better keyboard support

* Remove unrelated changes

* different fix

---------

Co-authored-by: Leon Sorokin <leeoniya@gmail.com>

dprokop committed:
AdHocFilters: Use queries in ad hoc filters api calls (#87624)

renovate[bot] committed:
Update dependency @types/lodash to v4.17.1 (#87419)

* Update dependency @types/lodash to v4.17.1

* fix grafana-data

* simpler grafana-data fix

* simpler code

* commit a betterer crime (it was already doing this implicitly before...)

---------

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>
Co-authored-by: Gábor Farkas <gabor.farkas@gmail.com>
Co-authored-by: Ashley Harrison <ashley.harrison@grafana.com>

aocenas committed:
Chore: Updated go.work.sum file (#87622)

* Updated file

* fix quote character issue in workspace check

---------

Co-authored-by: Todd Treece <todd.treece@grafana.com>

linoman committed:
SAML: add referemce to azure ad limitations (#87571)

* update the url for Azure AD limitations

* add warnings of using Azure AD with SAML

renovate[bot] committed:
Update dependency i18next to v23.11.4

renovate[bot] committed:
Update dependency semver to v7.6.2

ashharrison90 committed:
Chore: Upgrade redux (#86877)

* update packages

* fix type errors

* upgrade redux toolkit as well

* don't need eslint-disable command

* remove comment

* fix unit tests

* call rtk query selector directly

* remove unnecessary checks

konrad147 committed:
Alerting: Fix MatcherFilter onChange handling (#87511)

* Fix input onChange handler

* Remove unused variables

torkelo committed:
MixedDataSource: Support multi value data source variable that issues a query to each data source (#87586)

* Revert "Revert "MixedDataSource: Support multi value data source variable tha…"

This reverts commit 979d87f46f9bd7c5fed326a3fe2b700d906ec655.

* fix issue

renovate[bot] committed:
Update dependency @reduxjs/toolkit to v2.2.4

renovate[bot] committed:
Update dependency react-redux to v9.1.2

JoaoSilvaGrafana committed:
Chore: Remove gf-forms from `SubMenuItems` and `RangeMatcherEditor` (#87633)

renovate[bot] committed:
Update dependency @grafana/scenes to v4.19.0 (#87642)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
Update dependency @react-types/shared to v3.23.0 (#87643)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

aangelisc committed:
CloudMonitoring: Improve legacy query migrations (#86069)

* Match backend and correctly migrate metricType

Update tests

* Ensure project is migrated or set if not defined

* Improve migrations logic

aangelisc committed:
AzureMonitor: Refactor Log Analytics backend (#87429)

* Remove unneeded error check

* Refactor to reduce cyclomatic complexity

* Add util function for parsing resultformat

* Make use of util

* Remove unneeded types

* Move types to their own file

* Move getApiUrl to utils

* Move traces functions to separate file

- Add separate helper for building trace string queries

* Add helper for determining resources

* Add test for RetrieveResources

* Don't append twice

* Refactor tests

aangelisc committed:
Merge branch 'main' into jck/azBasicLogs-backend

# Conflicts:
#	pkg/tsdb/azuremonitor/loganalytics/azure-log-analytics-datasource.go
#	pkg/tsdb/azuremonitor/loganalytics/azure-log-analytics-datasource_test.go
#	pkg/tsdb/azuremonitor/loganalytics/utils.go
#	public/app/plugins/datasource/azuremonitor/components/ConfigEditor/MonitorConfig.tsx

aangelisc commented:
Closing this as I've broken things
